src/edit/CodeMirror.js

```diff
@@ -145,40 +145,47 @@ function registerEventHandlers(cm) {
     let dx = other.left - touch.left, dy = other.top - touch.top
     return dx * dx + dy * dy > 20 * 20
   }
-  on(d.scroller, "touchstart", e => {
-    if (!signalDOMEvent(cm, e) && !isMouseLikeTouchEvent(e) && !clickInGutter(cm, e)) {
-      d.input.ensurePolled()
-      clearTimeout(touchFinished)
-      let now = +new Date
-      d.activeTouch = {start: now, moved: false,
-                       prev: now - prevTouch.end <= 300 ? prevTouch : null}
-      if (e.touches.length == 1) {
-        d.activeTouch.left = e.touches[0].pageX
-        d.activeTouch.top = e.touches[0].pageY
+  on(d.scroller, "touchstart", function (e) {
+      // alert("touch start", e.touches)
+      if (!signalDOMEvent(cm, e) && !isMouseLikeTouchEvent(e) && !clickInGutter(cm, e)) {
+        d.input.ensurePolled()
+        clearTimeout(touchFinished)
+        var now = +new Date
+        d.activeTouch = {start: now, moved: false,
+                         prev: now - prevTouch.end <= 300 ? prevTouch : null}
+        if (e.touches.length == 1) {
+          d.activeTouch.left = e.touches[0].pageX
+          d.activeTouch.top = e.touches[0].pageY
+        } else {
+          d.activeTouch.left = 0
+          d.activeTouch.top = 0
+        }
       }
-    }
-  })
-  on(d.scroller, "touchmove", () => {
-    if (d.activeTouch) d.activeTouch.moved = true
-  })
-  on(d.scroller, "touchend", e => {
-    let touch = d.activeTouch
-    if (touch && !eventInWidget(d, e) && touch.left != null &&
-        !touch.moved && new Date - touch.start < 300) {
-      let pos = cm.coordsChar(d.activeTouch, "page"), range
-      if (!touch.prev || farAway(touch, touch.prev)) // Single tap
-        range = new Range(pos, pos)
-      else if (!touch.prev.prev || farAway(touch, touch.prev.prev)) // Double tap
-        range = cm.findWordAt(pos)
-      else // Triple tap
-        range = new Range(Pos(pos.line, 0), clipPos(cm.doc, Pos(pos.line + 1, 0)))
-      cm.setSelection(range.anchor, range.head)
-      cm.focus()
-      e_preventDefault(e)
-    }
-    finishTouch()
-  })
-  on(d.scroller, "touchcancel", finishTouch)
+      finishTouch()
+    });
+    // on(d.scroller, "touchmove", function () {
```

Comment:
Removing this will not impact another scenario? Please make sure all the scenarios will continue to work.